frontend/plugins/frontend-tools/OutputTimer.cpp

```diff
@@ -0,0 +1,358 @@
+#include "OutputTimer.hpp"
+
+#include <obs-module.h>
+#include <qt-wrappers.hpp>
+
+#include <QMainWindow>
+#include <QPushButton>
+#include <QTimer>
+
+#include "ui_OutputTimer.h"
+#include "moc_OutputTimer.cpp"
+
+namespace {
+QScopedPointer<OutputTimer> dialog;
+
+int getInterval(int hours, int minutes, int seconds)
+{
+	return ((hours * 3600) + (minutes * 60) + seconds) * 1000;
+}
+
+void updateButtonState(QPushButton *button, bool active)
+{
+	button->setText(active ? obs_module_text("Stop") : obs_module_text("Start"));
+	setClasses(button, active ? "state-active" : "");
+}
+
+QString formatTime(int ms)
+{
+	int remainingTime = ms / 1000;
+
+	int seconds = remainingTime % 60;
+	int minutes = (remainingTime % 3600) / 60;
+	int hours = remainingTime / 3600;
+
+	return QString::asprintf("%02d:%02d:%02d", hours, minutes, seconds);
+}
+} // namespace
+
+OutputTimer::OutputTimer(QWidget *parent) : QDialog(parent), ui(new Ui_OutputTimer)
+{
+	ui->setupUi(this);
+
+	setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+	obs_frontend_add_save_callback(onSave, this);
+	obs_frontend_add_event_callback(onEvent, this);
+
+	ui->streamStoppingIn->hide();
+	ui->streamTimeLeft->hide();
+	ui->recordStoppingIn->hide();
+	ui->recordTimeLeft->hide();
+
+	connect(ui->buttonBox->button(QDialogButtonBox::Close), &QPushButton::pressed, this, &OutputTimer::hide);
+}
+
+OutputTimer::~OutputTimer()
+{
+	obs_frontend_save();
+
+	obs_frontend_remove_save_callback(onSave, this);
+	obs_frontend_remove_event_callback(onEvent, this);
+}
+
+void OutputTimer::updateLabels()
+{
+	if (streamTimer)
+		ui->streamTimeLeft->setText(formatTime(streamTimer->remainingTime()));
+	else
+		ui->recordTimeLeft->setText(formatTime(getStreamInterval()));
```

Comment:
I _think_ a copy paste typo might have snuck in here:
`ui->recordTimeLeft->setText(formatTime(getStreamInterval()));`
... vs ...
`ui->streamTimeLeft->setText(formatTime(getStreamInterval()));`